File: web/src/types/schedule.ts
export interface Schedule {
  id: string
  stackId: string
  name: string
  cronExpr: string
  duration: number
  timezone: string
  enabled: boolean
  createdAt: string
  updatedAt: string
}

export interface QueuedUpdate {
  id: string
  stackId: string
  scheduleId: string | null
  composeContent: string
  commitMessage: string
  status: 'queued' | 'deploying' | 'deployed' | 'failed' | 'cancelled'
  queuedAt: string
  deployAfter: string | null
  deployedAt: string | null
}

export interface CreateScheduleRequest {
  name: string
  cronExpr: string
  duration: number
  timezone: string
  enabled: boolean
}

export interface UpdateScheduleRequest {
  name?: string
  cronExpr?: string
  duration?: number
  timezone?: string
  enabled?: boolean
}

export interface QueueUpdateRequest {
  composeContent: string
  commitMessage?: string
  scheduleId?: string
}

File: internal/models/schedule.go
package models

type Schedule struct {
	ID        string `db:"id" json:"id"`
	StackID   string `db:"stack_id" json:"stackId"`
	Name      string `db:"name" json:"name"`
	CronExpr  string `db:"cron_expr" json:"cronExpr"`
	Duration  int    `db:"duration" json:"duration"`
	Timezone  string `db:"timezone" json:"timezone"`
	Enabled   bool   `db:"enabled" json:"enabled"`
	CreatedAt Time   `db:"created_at" json:"createdAt"`
	UpdatedAt Time   `db:"updated_at" json:"updatedAt"`
}

type QueuedUpdate struct {
	ID             string  `db:"id" json:"id"`
	StackID        string  `db:"stack_id" json:"stackId"`
	ScheduleID     *string `db:"schedule_id" json:"scheduleId"`
	ComposeContent string  `db:"compose_content" json:"composeContent"`
	CommitMessage  string  `db:"commit_message" json:"commitMessage"`
	Status         string  `db:"status" json:"status"`
	QueuedAt       Time    `db:"queued_at" json:"queuedAt"`
	DeployAfter    *Time   `db:"deploy_after" json:"deployAfter"`
	DeployedAt     *Time   `db:"deployed_at" json:"deployedAt"`
}
